xconn/session.py

```diff
@@ -142,6 +147,24 @@ def publish(self, topic: str, args: list[Any] = None, kwargs: dict = None, optio
 
         self.base_session.send(data)
 
+    def leave(self):
+        self.goodbye_request = Future()
+
+        goodbye = messages.Goodbye(messages.GoodbyeFields({}, uris.CLOSE_REALM))
+        data = self.session.send_message(goodbye)
+        self.base_session.send(data)
+
+        timeout = 10
+        start_time = time.time()
+
+        while not self.goodbye_request.done():
```

Comment:
no need to do this. The `self.goodbye_request.result()` takes timeout as parameter, just provide it there.